lib/Charts/ChartRenderer.js

```diff
@@ -575,26 +772,50 @@ function highlightAndTooltip(
   fireClick = false
 ) {
   if (!hasData) {
+    Tooltip.hide(tooltip);
     return;
   }
   const localCoords = d3Mouse(g.nodes()[0]);
-  const hoverX = scales.x.invert(localCoords[0]);
+  const hoverX = scales.zoomedX
+    ? scales.zoomedX.invert(localCoords[0])
+    : scales.x.invert(localCoords[0]);
+
   const selectedData = findSelectedData(data, hoverX);
-  hilightData(selectedData, scales, g);
+  if (selectedData.length === 0) {
+    highlightXAxis(hoverX, scales, g);
+  } else {
+    highlightXAxis(selectedData[0].point.x, scales, g);
+  }
+
   const chartBoundingRect = g["_parents"][0].getBoundingClientRect(); // Strangely, g's own width can sometimes be far too wide.
+  if (selectedData.length === 0) {
+    Tooltip.hide(tooltip);
+    return;
+  }
   Tooltip.show(
-    Tooltip.html(selectedData),
+    Tooltip.html(selectedData, hoverX),
     tooltip,
     state.tooltipSettings,
     chartBoundingRect
   );
 
   if (fireClick) {
-    selectedData.forEach(chartData => {
-      if (chartData.onClick) {
-        chartData.onClick(hoverX); // TODO also pass Y
-      }
-    });
+    // Because double click should be used for zooming the chart out
```

Comment:
I don't quite understand the need for this manual double click detection. D3 lets you attach any event to a DOM element with `.on`, right? And browsers have a `dblclick` event. So I don't see why `whatever.on("dblclick", callback)` wouldn't work. One good reason to avoid manually detecting double clicks is that its an accessibility issue. 250ms is ok for some users, but not all. That's why it's configurable at the OS level.

Reply:
So I think the problem I was trying to solve here was that a `dblclick` was supposed to make the chart zoom out. However it was also triggering the single `click` event which meant it was triggering a change in imagery.

Reply:
Ah I see. Yeah that makes it awkward. It'd be slightly better to make the dblclick event cancel the delayed action from the click event, rather than letting a second click (whether interpreted as a double click or not) cancel the action. Even better would be to do the click action immediately, and then "undo" it when the dblclick arrives, but that might be hard. And even better than all that is to not rely on double-clicking to zoom out, cause I'm not sure anyone is going to think of that anyway. :)